beacon_node/lighthouse_network/src/rpc/mod.rs

```diff
@@ -408,16 +405,29 @@ where
                                     format!("Wait {:?}", wait_time).into(),
                                 ),
                             );
+                            return;
                         }
+                        // No rate limiting, continue.
+                        Ok(_) => {}
                     }
-                } else {
-                    // No rate limiting, send the event to the user
-                    self.events.push(ToSwarm::GenerateEvent(RPCMessage {
+                }
+                // If we received a Ping, we queue a Pong response.
+                if let InboundRequest::Ping(_) = req {
```

Comment:
I don't quite get the need to handle Ping differently in the rpc behaviour. What was the issue with the way we handled it before from lighthouse_network service?

Reply:
wdym differently? The issue before is that Replying to `Ping`s with `Pong`s is part of the [spec](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#ping) It doesn't make sense to  handle Pong responses at the `Swarm` level instead of at the Behaviour level both in terms of [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns) and latency.  This is also how the [`Ping`](https://docs.rs/libp2p/latest/libp2p/ping/index.html) behaviour operates.

Reply:
My point is that we are separating a specific concern (replying to the ping message). But we are still handling other requirements of the spec at the swarm level (sending a metadata request if the seq number has changed, setting a timer for the next ping request). 

But I do see your point that replying to Pings fits somewhat better in the behaviour.

Reply:
thanks for expanding Pawan, yeah that is right but to that we cannot do anything right? as it escapes `Rpc`'s scope what updates the metada's sequence number.